pkg/backend/openshiftcluster.go

```diff
@@ -76,7 +77,21 @@ func (ocb *openShiftClusterBackend) handle(ctx context.Context, log *logrus.Entr
 	stop := ocb.heartbeat(ctx, cancel, log, doc)
 	defer stop()
 
-	m, err := openshiftcluster.NewManager(log, ocb.env, ocb.db.OpenShiftClusters, ocb.billing, doc)
+	r, err := azure.ParseResourceID(doc.OpenShiftCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	subscriptionDoc, err := ocb.db.Subscriptions.Get(ctx, r.SubscriptionID)
+	if err != nil {
+		return err
+	}
+
+	if subscriptionDoc == nil {
+		return fmt.Errorf("Subscription document for '%s' not found", r.SubscriptionID)
+	}
```

Comment:
I gave you a bad steer - the Subscriptions.Get contract is to give you a document or an error.  Turns out this case can't happen.